src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,476 @@
+    g = gravity_constant(equations)
+    (; lambda, bathymetry_type) = equations
+    (; D1) = solver
+
+    # `q` and `dq` are `ArrayPartition`s. They collect the individual
+    # arrays for the total water height `eta = h + b`, the velocity `v`,
+    # and the additional variables `w` and `H`.
+    eta, v, D, w, H = q.x
+    dh, dv, dD, dw, dH = dq.x # dh = deta since b is constant in time
+    fill!(dD, zero(eltype(dD)))
+
+    @trixi_timeit timer() "hyperbolic terms" begin
+        # Compute all derivatives required below
+        (; h, b, b_x, H_over_h, h_x, v_x, hv_x, v2_x, h_hpb_x, H_x, H2_h_x, w_x, hvw_x, tmp) = cache
+
+        @. b = equations.eta0 - D
+        @. h = eta - b
+        if !(bathymetry_type isa BathymetryFlat)
+            mul!(b_x, D1, b)
+        end
+
+        # h_x = D1 * h
+        mul!(h_x, D1, h)
+
+        # v_x = D1 * v
+        mul!(v_x, D1, v)
+
+        # hv2_x = D1 * (h * v)
+        @. tmp = h * v
+        mul!(hv_x, D1, tmp)
+
+        # v2_x = D1 * (v.^2)
+        @. tmp = v^2
+        mul!(v2_x, D1, tmp)
+
+        # h_hpb_x = D1 * (h .* eta)
+        @. tmp = h * eta
+        mul!(h_hpb_x, D1, tmp)
+
+        # H_x = D1 * H
+        mul!(H_x, D1, H)
+
+        # H2_h_x = D1 * (H^2 / h)
+        @. H_over_h = H / h
+        @. tmp = H * H_over_h
+        mul!(H2_h_x, D1, tmp)
+
+        # w_x = D1 * w
+        mul!(w_x, D1, w)
+
+        # hvw_x = D1 * (h * v * w)
+        @. tmp = h * v * w
+        mul!(hvw_x, D1, tmp)
+
+        # Plain: h_t + (h v)_x = 0
+        #
+        # Split form for energy conservation:
+        # h_t + h_x v + h v_x = 0
+        @. dh = -(h_x * v + h * v_x)
+
+        # Plain: h v_t + h v v_x + g (h + b) h_x
+        #              + ... = 0
+        #
+        # Split form for energy conservation:
+        # h v_t + g (h (h + b))_x - g (h + b) h_x
+        #       + 1/2 h (v^2)_x - 1/2 v^2 h_x  + 1/2 v (h v)_x - 1/2 h v v_x
+        #       + λ/6 H^2 / h^2 h_x + λ/3 H_x - λ/3 H/h H_x - λ/6 (H^2 / h)_x
+        #       + λ/2 b_x - λ/2 H/h b_x = 0
+        lambda_6 = lambda / 6
+        lambda_3 = lambda / 3
+        lambda_2 = lambda / 2
+        if bathymetry_type isa BathymetryFlat
+            @. dv = -(g * h_hpb_x - g * (h + b) * h_x
+                      +
+                      0.5 * h * v2_x - 0.5 * v^2 * h_x
+                      +
+                      0.5 * hv_x * v - 0.5 * h * v * v_x
+                      + lambda_6 * (H_over_h * H_over_h * h_x - H2_h_x)
+                      + lambda_3 * (1 - H_over_h) * H_x) / h
+        else
+            @. dv = -(g * h_hpb_x - g * (h + b) * h_x
+                      +
+                      0.5 * h * v2_x - 0.5 * v^2 * h_x
+                      +
+                      0.5 * hv_x * v - 0.5 * h * v * v_x
+                      + lambda_6 * (H_over_h * H_over_h * h_x - H2_h_x)
+                      + lambda_3 * (1 - H_over_h) * H_x
+                      + lambda_2 * (1 - H_over_h) * b_x) / h
+        end
```

Comment:
```suggestion
         @. dv = -(g * h_hpb_x - g * (h + b) * h_x
                   +
                   0.5 * h * v2_x - 0.5 * v^2 * h_x
                   +
                   0.5 * hv_x * v - 0.5 * h * v * v_x
                   + lambda_6 * (H_over_h * H_over_h * h_x - H2_h_x)
                   + lambda_3 * (1 - H_over_h) * H_x) / h
        if !(bathymetry_type isa BathymetryFlat)
            @. dv += (lambda_2 * (1 - H_over_h) * b_x) / h
        end
```
Could we make this shorter to reduce code duplication?